pandas/core/base.py

```diff
@@ -602,9 +602,9 @@ def _aggregate_multiple_funcs(self, arg, _level, _axis):
         if not len(results):
             raise ValueError("no results")
 
-        try:
+        if all(np.ndim(x) > 0 for x in results):
             return concat(results, keys=keys, axis=1, sort=False)
-        except TypeError:
```

Comment:
What's an example that now ran into this TypeError?

Reply:
> What's an example that now ran into this TypeError?

`results = [3, 2, 1.5]; keys = ['size', 'count', 'mean']`

Reply:
Sorry, I meant the original dataframe / groupby example or test case that gives those values at this location

Reply:
Identifying some relevant cases with an `assert False`, the place to look is in tests.frame.test_apply, e.g. test_non_callable_aggregates